### DISCUSSION TITLE
Cypress on quasar v1?

### DISCUSSION THREAD

**Discussion by mtn090899** [quasarframework/quasar#12420] [Quasar App Extensions]
Recently wanted to do some testing on an app built on quasar v1 a couple years ago, but adding cypress is not working in the slightest. It is asking for a v2 and above version for quasar. Is there any way I can still run and work with cypress on this application? (Even running quasar upgrade didn't help, the app was built by others in quasar v1 so I have no idea how to go about this, but I do need to get cypress to work on it). 

Any advice would be helpful, thank you!

**Comment by MilosPaunovic** [comment]
Check this out:
[https://github.com/quasarframework/quasar-testing/tree/dev/packages/testing](https://github.com/quasarframework/quasar-testing/tree/dev/packages/testing)

Way to add `Cypress` to `Q1` project is with `quasar ext add @quasar/testing@qv1`.

**Reply by mtn090899** [reply]
Thank you so much! I seriously could not find it for my life.

**Reply by mtn090899** [reply]
I apologize to further elaborate on this question, however upon installing @quasar/testing@qv1, and adding cypress, I receive an error still asking for the same thing during the installation, to have a later version of Quasar. So when running the command quasar test --e2e cypress it is unknown due to the failure to install e2e-cypress. Am I doing something wrong?

**Reply by MilosPaunovic** [reply]
Have you removed `node_modules` and `package-lock.json` before running that command, just to be safe?

I have this `Q1` starter template with Cypress installed:
[https://github.com/MilosPaunovic/quasar-v1-starter](https://github.com/MilosPaunovic/quasar-v1-starter)

So maybe you can give it a look and compare, potentially you'll find what's missing, if not, please create a minimal reproduction repo so someone could give it a closer look.
